src/QA/SoftMocks.php

```diff
@@ -949,16 +949,60 @@ public static function callFunction($namespace, $func, $params)
         return call_user_func_array($func, $params);
     }
 
-    public static function callNew($class, $args)
+    public static function callNew($class, $args, $scope)
     {
         if (isset(self::$new_mocks[$class])) {
             return call_user_func_array(self::$new_mocks[$class], $args);
         }
 
         $Rc = new \ReflectionClass($class);
-        return $Rc->newInstanceArgs($args);
+        $Constructor = $Rc->getConstructor();
+
+        if ($Constructor && self::isConstructorCanBeForced($Constructor, $scope)) {
+            $instance = $Rc->newInstanceWithoutConstructor();
+            $Constructor->setAccessible(true);
+            $Constructor->invokeArgs($instance, $args);
+            $Constructor->setAccessible(false);
+        } else {
+            $instance = $Rc->newInstanceArgs($args);
+        }
+
+        return $instance;
     }
 
+    private static function isConstructorCanBeForced(\ReflectionMethod $constructor, $scope)
+    {
+        if ($constructor->isPublic()) {
+            return false;
+        }
+
+        $class = $constructor->getDeclaringClass();
+        if ($constructor->isPrivate() && $class->getName() === $scope) {
+            return true;
+        }
+
+        if ($constructor->isProtected() && self::isSubclassOf(new \ReflectionClass($scope), $class)) {
```

Comment:
I will answer my own question. The code above does work. Still I would like to have some evidence that this behaviour is documented and that it is not going to change in future versions of PHP or even HHVM.